src/BlazorWebView/src/WindowsForms/WindowsFormsDispatcher.cs

```diff
@@ -72,8 +72,23 @@ public override async Task InvokeAsync(Func<Task> workItem)
 				}
 				else
 				{
-					var asyncResult = _dispatchThreadControl.BeginInvoke(workItem);
-					await Task.Factory.FromAsync(asyncResult, _dispatchThreadControl.EndInvoke);
+					var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+					// BeginInvoke specifically expects an `Action` so avoid using var here.
+					Action action = async () =>
+					{
+						try
+						{
+							await workItem();
+							tcs.TrySetResult();
+						}
+						catch (Exception ex)
+						{
+							tcs.TrySetException(ex);
+						}
+					};
+
+					var asyncResult = _dispatchThreadControl.BeginInvoke(action, workItem, tcs);
```

Comment:
Like the issue noted, `BeginInvoke` does not await the operation. Consequently in the current format, the changes to `Index` would result in an exception. Building on the pattern that @noseratio suggested in the issue, we can await both the result of `BeginInvoke` and the completion of the operation.